### PR TITLE
[DEVOPS-248] Migrate CircleCI to Github actions

### PR DESCRIPTION
**Pull request by MarkCalvert**
## Motivation and Context

**JIRA issue:** https://salsadigital.atlassian.net/browse/DEVOPS-248

## Changed

1. Created new Github action workflows based off current CircleCI implementation

### Screenshots

Screenshot 2023-08-15 at 11 00 39 AM

## How Has This Been Tested?

- Tested Github action workflow commit
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist

- [ ] I've added relevant changes to the documentation.
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] My change requires a template update for create-ripple-app.
- [ ] I have added template update script for next release.

### CODE REVIEW

File: .github/workflows/nightwatch.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      e2e_project:
+        type: string
+        required: true
+      e2e_be_url:
+        type: string
+      e2e_fe_url:
+        type: string
+jobs:
+  e2e_be:
+    if: startsWith(github.ref_name , 'release/') && inputs.e2e
```
Comment (tim-yao) [inline .github/workflows/nightwatch.yml:17]: Hi @MarkCalvert please remove this release branch condition, it's not required anymore. Cheers!
```suggestion
    if: inputs.e2e
```

File: .github/workflows/nightwatch.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    secrets: inherit
+    with:
+      tags: "(@core or @${{ inputs.e2e_project }}) and @regression"
+      be_url: ${{ inputs.e2e_be_url }}
+      project: ${{ inputs.e2e_project }}
+      browser: 'chrome'
+      test_id: 'fixture'
+      test_type: ':nightwatch:'
+  e2e_fe_smoke:
+    if: startsWith(github.ref_name , 'release/') && inputs.e2e
```
Comment (tim-yao) [inline .github/workflows/nightwatch.yml:28]: Same as this and all e2e jobs. Thanks!

**Comment by tim-yao** [comment]
@MarkCalvert @FleetAdmiralButter The ci trigger is not working:
```
 Triggering end to end testing for ripple...
Error: No E2E GitHub token found, end to end is not triggered. Please make sure E2E_GITHUB_TOKEN is set up.
```
Do we need to wait for the middleware to work first then we can test it?

**Comment by tim-yao** [comment]
> 

@MarkCalvert ok. Can you please use your own GitHub token to call the same curl request from your local? Then we can see if it works.

**Comment by MarkCalvert** [comment]
Hi @tim-yao,
Using the GitHub CLI tool, I found a workaround to trigger the nightwatch workflow for testing.
Here is the gh CLI command you can run from the locally cloned repo.
`gh workflow run nightwatch.yml --ref feature/migrating-circleci-to-github-actions -f e2e=true -f e2e_project=reference -f e2e_be_url=https://develop.content.reference.sdp.vic.gov.au -f e2e_fe_url=https://app.pr-1313.ripple.sdp4.sdp.vic.gov.au/ `

You can view the actions I have triggered for the nightwatch workflow and will notice they fail in the e2e reusable workflow tests, can you please review these?

**Comment by MarkCalvert** [comment]
@tim-yao I have disabled the workflows by updating the on event branches to a name that does not exist, so they will not be triggered on master and uat pushes.
https://github.com/dpc-sdp/ripple/pull/1313/commits/06c07d932d6e3c86a8dfc10dee4bfe4fff461f5d

**Pull request merged**